homeassistant/components/openalpr.py

```diff
@@ -0,0 +1,440 @@
+    runtime = config[DOMAIN].get(CONF_RUNTIME)
+    use_render_fffmpeg = False
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    openalpr_device = []
+
+    for device in config[DOMAIN].get(CONF_ENTITIES):
+        input_source = device.get(CONF_INPUT)
+        render = device.get(CONF_RENDER)
+
+        ##
+        # create api
+        if engine == ENGINE_LOCAL:
+            alpr_api = OpenalprApiLocal(
+                runtime=runtime,
+                region=region,
+            )
+        else:
+            alpr_api = OpenalprApiCloud(
+                api_key=api_key,
+                region=region,
+            )
+
+        ##
+        # Create Alpr device / render engine
+        if render == RENDER_FFMPEG:
+            use_render_fffmpeg = True
+            if not run_test(input_source):
+                _LOGGER.error("'%s' is not valid ffmpeg input", input_source)
+                continue
+
+            alpr_dev = OpenalprDeviceFFmpeg(
+                name=device.get(CONF_NAME),
+                interval=device.get(CONF_INTERVAL),
+                api=alpr_api,
+                input_source=input_source,
+                extra_arguments=device.get(CONF_EXTRA_ARGUMENTS),
+            )
+        else:
+            alpr_dev = OpenalprDeviceImage(
+                name=device.get(CONF_NAME),
+                interval=device.get(CONF_INTERVAL),
+                api=alpr_api,
+                input_source=input_source,
+                username=device.get(CONF_USERNAME),
+                password=device.get(CONF_PASSWORD),
+            )
+
+        # register shutdown event
+        openalpr_device.append(alpr_dev)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, alpr_dev.shutdown)
+
+    component.add_entities(openalpr_device)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    def _handle_service_scan(service):
+        """Handle service for immediately scan."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.scan()
+
+    hass.services.register(DOMAIN, SERVICE_SCAN,
+                           _handle_service_scan,
+                           descriptions[DOMAIN][SERVICE_SCAN],
+                           schema=SERVICE_SCAN_SCHEMA)
+
+    # Add restart service only if a device use ffmpeg as render
+    if not use_render_fffmpeg:
+        return True
+
+    def _handle_service_restart(service):
+        """Handle service for restart ffmpeg process."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.restart()
+
+    hass.services.register(DOMAIN, SERVICE_RESTART,
+                           _handle_service_restart,
+                           descriptions[DOMAIN][SERVICE_RESTART],
+                           schema=SERVICE_RESTART_SCHEMA)
+
+    return True
+
+
+class OpenalprDevice(Entity):
```

Comment:
You're defining entities but you're always hiding the entities and do not even have a state defined for any of them. It doesn't sound to me like you want to actually have an entity.

I would consider this platform to be implemented like a sensor that will always show the most prominent license plate in the frame but only if certainty is > 70% or something.